testing/test_pdb.py

```diff
@@ -954,3 +954,30 @@ def test_2():
     rest = child.read().decode("utf8")
     assert "no tests ran" in rest
     TestPDB.flush(child)
+
+
+def test_pdb_quit_after_debug(testdir):
```

Comment:
Strange, I get the same behavior as this test shows even in current `features`:

```
λ pytest .tmp\test_quit_db.py
======================================= test session starts =======================================
...
collected 1 item

.tmp\test_quit_db.py
>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB set_trace (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>
--Return--
> c:\pytest\.tmp\test_quit_db.py(6)test_1()->None
-> __import__('pdb').set_trace()
(Pdb) debug foo
ENTERING RECURSIVE DEBUGGER
> <string>(1)<module>()->None
((Pdb)) q
LEAVING RECURSIVE DEBUGGER
(Pdb) q
Exit: Quitting debugger


!!!!!!!!!!!!!!!!!!!!!!!!!!!! _pytest.outcomes.Exit: Quitting debugger !!!!!!!!!!!!!!!!!!!!!!!!!!!!!
================================== no tests ran in 6.57 seconds ===================================
```

What do you get when you execute this test step by step on Linux?

Reply:
Oh, that's related to pdbpp again - similar to https://github.com/pytest-dev/pytest/pull/4347 I guess.

pdb uses a new Pdb instance inside of `do_debug`, and pdbpp injects its custom `Pdb` into the function, which results in its / pytest's `set_quit` being called then.

Will investigate / adjust.